File: C++/minimum-sensors-to-cover-grid.cpp
// Time:  O(1)
// Space: O(1)

// math
class Solution {
public:
    int minSensors(int n, int m, int k) {
        const auto& ceil_divide = [](int a, int b) {
            return (a + b - 1) / b;
        };

        return static_cast<int64_t>(ceil_divide(n, 2 * k + 1)) * ceil_divide(m, 2 * k + 1);
    }
};

File: Python/minimum-sensors-to-cover-grid.py
# Time:  O(1)
# Space: O(1)

# math
class Solution(object):
    def minSensors(self, n, m, k):
        """
        :type n: int
        :type m: int
        :type k: int
        :rtype: int
        """
        def ceil_divide(a, b):
            return (a+b-1)//b
    
        return ceil_divide(n, 2*k+1)*ceil_divide(m, 2*k+1)
